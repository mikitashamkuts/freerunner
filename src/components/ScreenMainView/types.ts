import {CustomComponentProps} from '../../types';

export interface Props extends CustomComponentProps {
  isWithAnimation?: boolean;
}
