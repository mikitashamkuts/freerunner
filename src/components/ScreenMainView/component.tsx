import React, {FC, memo, useEffect, useRef} from 'react';
import {Animated} from 'react-native';

import {whyDidItRenderConfig} from '../../../debug';
import {defaultAnimationDuration} from '../../constants';
import {useTheme} from '../../hooks';

import {Props, styles} from '.';

const ScreenMainView: FC<Props> = ({children, isWithAnimation = true}) => {
  const themedStyles = useTheme(styles);
  const fadeAnim = useRef(new Animated.Value(isWithAnimation ? 0 : 1)).current;

  useEffect(() => {
    isWithAnimation &&
      Animated.timing(fadeAnim, {
        toValue: 1,
        duration: defaultAnimationDuration,
        useNativeDriver: true,
      }).start();
  }, [fadeAnim, isWithAnimation]);

  return (
    <Animated.View
      style={[themedStyles.container, {opacity: fadeAnim, justifyContent: 'space-between'}]}>
      {children}
    </Animated.View>
  );
};

ScreenMainView.whyDidYouRender = whyDidItRenderConfig.UIComponentDebugActive;

export default memo(ScreenMainView);
