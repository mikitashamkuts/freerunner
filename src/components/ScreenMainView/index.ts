export {default as ScreenMainView} from './component';
export {styles} from './styles';
export * from './types';
