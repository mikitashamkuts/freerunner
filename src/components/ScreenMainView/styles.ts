import {backgroundColorList} from '@src/design';
import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    backgroundColor: backgroundColorList.Light,
    flex: 1,
    justifyContent: 'space-between',
  },
  dark_container: {
    backgroundColor: backgroundColorList.Dark,
  },
});
