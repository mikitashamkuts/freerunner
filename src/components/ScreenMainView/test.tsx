/**
 * @format
 */

import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';
import React from 'react';
import {Text} from 'react-native';
import ScreenMainView from './component';

// Mock the useTheme hook to return a static style
jest.mock('../../hooks', () => ({
  useTheme: () => ({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: 'white',
    },
  }),
}));

// Mock the whyDidItRenderConfig to avoid issues with the debugging configuration
jest.mock('../../../debug', () => ({
  whyDidItRenderConfig: {
    UIComponentDebugActive: false,
  },
}));

describe('ScreenMainView Component', () => {
  it('Renders with children and matches snapshot (no animation) : ', () => {
    const tree = renderer
      .create(
        <ScreenMainView isWithAnimation={false}>
          <Text>Test Child</Text>
        </ScreenMainView>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders without children and matches snapshot (no animation) : ', () => {
    const tree = renderer.create(<ScreenMainView isWithAnimation={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
