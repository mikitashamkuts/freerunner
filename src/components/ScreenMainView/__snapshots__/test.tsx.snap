// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenMainView Component Renders with children and matches snapshot (no animation) :  1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text>
    Test Child
  </Text>
</View>
`;

exports[`ScreenMainView Component Renders without children and matches snapshot (no animation) :  1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
/>
`;
